tests/webTables.spec.ts

```diff
@@ -0,0 +1,182 @@
+    await page.waitForSelector('tbody')
+
+    const allRows = await page.locator('tbody tr').all()
+    // Iterate through rows to find 'surgery'
+    for (let row of allRows) {
+        if (await row.locator('input').inputValue() === 'surgery') {
+            // ... click 'Edit'
+            await row.getByRole('button', { name: 'Edit' }).click()
+            break;
+        }
+    }
+    //6. Add assertion "Edit Specialty" page is displayed
+    await expect(page.locator('h2')).toHaveText('Edit Specialty')
+    //7. Update the specialty from "surgery" to "dermatology" and click "Update button"
+    //8. Add assertion that "surgery" was changed to "dermatology" in the list of specialties
+    const specialtyInput = page.locator('#name')
+    await expect(specialtyInput).toHaveValue('surgery')
+    await specialtyInput.clear()
+    await specialtyInput.fill('dermatology')
+    await page.getByText('Update').click()
+    //9. Select the VETERINARIANS menu item in the navigation bar, then select "All"
+    await page.getByText("Veterinarians").click()
+    await page.getByText("All").click()
+    //10. On the Veterinarians page, add assertion that "Rafael Ortega" has specialty "dermatology"
+    await expect(page.getByRole('row', { name: 'Rafael Ortega' }).getByRole('cell').nth(1)).toHaveText('dermatology')
+    //11. Navigate to SPECIALTIES page, revert the changes renaming "dermatology" back to "surgery"
+    await page.getByRole('link', { name: 'Specialties' }).click()
+
+    for (let row of allRows) {
+        if (await row.locator('input').inputValue() === 'dermatology') {
+            await row.getByRole('button', { name: 'Edit' }).click()
+            break;
+        }
+    }
+
+    await expect(specialtyInput).toHaveValue('dermatology')
+    await specialtyInput.clear()
+    await specialtyInput.fill('surgery')
+    await page.getByText('Update').click()
+})
+
+test('validate specialty lists', async ({ page }) => {
+    //1. Select the SPECIALTIES menu item in the navigation bar
+    await page.getByText('Specialties').click()
+    await page.waitForSelector('tbody')
+    //2. On the Specialties page, select "Add" button. Type the new specialty "oncology" and click "Save" button
+    await page.getByRole('button', { name: 'Add' }).click()
+    await page.locator('#name').fill('oncology')
+    await page.getByText('Save').click()
+    await expect(page.locator('tbody tr').last().getByRole('textbox')).toHaveValue('oncology')
+    //3. Extract all values of specialties and put them into the array.
+    const allSpecialtyInputs = await page.locator('tbody tr').getByRole('textbox').all()
+    const allSpecialties: string[] = []
+
+    for (const input of allSpecialtyInputs) {
+        allSpecialties.push(await input.inputValue())
+    }
+
+    //4. Select the VETERINARIANS menu item in the navigation bar, then select "All"
+    await page.getByText('Veterinarians').click()
+    await page.locator(".dropdown-menu").getByText(" All").click()
```

Comment:
Earlier in your code in line 122, you have used a simpler version of the locator than this version. Why?